### PR TITLE
Try: Polish labels and consolidate options in preferences.

### PR DESCRIPTION
**Pull request by jasmussen**
Followup to https://github.com/WordPress/gutenberg/pull/32922#issuecomment-867066725.

## Description

In this PR I've taken a stab at consolidating options in the Preferences dialog, as well as simplify the titles and add descriptive text instead. 

_I'm marking this one as a draft and in progress, as I expect to tweak the copy and particulars based on feedback._

Before:

Screenshot 2021-07-01 at 13 11 54

After:

Screenshot 2021-07-01 at 13 18 31

Notice how I've merged options from "Appearance" into the "General" section, as all of those options are conceptually related. I'm also not sure it's useful to have an entire section dedicated to just two options — "Display button labels" and "Use theme styles". While it's possible we'll have more options in the future, it doesn't seem that useful to optimize for that until we get there.

Before:

Screenshot 2021-07-01 at 13 20 01

After:

Screenshot 2021-07-01 at 13 20 41

Let me know your thoughts.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by javierarce** [comment]
I like those changes, and I think moving "Appearances" to the General section makes things straightforward.

I just found a typo in "Shows text instead of icons in toolbar."  (the "the" before toolbar is missing).

**Comment by jasmussen** [comment]
Thank you, I fixed the typo.

**Comment by stokesman** [comment]
Regarding that help text "Shows text instead of icons in the toolbar." Because it affects more than one toolbar, I think the singular "toolbar" doesn't seem right. It could be made plural or maybe the text could reduce to "Shows text instead of icons." because "Display button labels" already specifies the context.
<hr>

I think this is much better! We could call it an iteration and merge it. I also think there may still be room for improvement. So I'll share my thoughts but **I don't think any of them need to be addressed by this PR**.

Mainly, I have wondered (since well before this PR) if the settings under "Block Interactions" are where they should be and think they may be equally or more expected to appear under a section of "General". Here's why:
- "Contain text cursor inside block" seems conceptually very similar to "Spotlight mode" and an important setting we'd want to featured more prominently.
- "Show most used blocks" could be considered an interface setting since it affects the library/main inserter.

If that seems worth considering here's one take on it:

<details>
<summary>"General" tab reorganized with absorption of settings from "Block interactions"</summary>

- Editing
  - Include pre-publish checklist
  - Contain text cursor inside block
  - Spotlight mode
  - Use theme styles (here because it affects the canvas/blocks and not the interface)
- Interface (perhaps a better title than Appearance?)
  - Display button labels
  - Show most used blocks
  - Reduce the interface
  - Display block breadcrumbs

</summary>

**Comment by jasmussen** [comment]
Good thoughts, thanks for the review!

> "Contain text cursor inside block" seems conceptually very similar to "Spotlight mode" and an important setting we'd want to featured more prominently.

I'm hesitant to move the entire "Block interactions" feature into the General section, notably because the contain feature pretty drastically changes the writing flow. Specifically, it stops the arrow-key navigation between blocks:

contain

This is a useful feature for some, but notably not all people. Some screen reader users prefer the caret stops at the end of the block, others might prefer it to have a more page buildery experience. But the feature can also easily be misunderstood and cause frustration when the normal text-editor arrow-key navigation stops working. I wonder if we could expand or rephrase the help text to be clearer?

**Comment by stokesman** [comment]
>Some screen reader users prefer the caret stops at the end of the block, others might prefer it to have a more page buildery experience. But the feature can also easily be misunderstood and cause frustration when the normal text-editor arrow-key navigation stops working.

It surprises me that the feature could be easily misunderstood. Especially given the help text. Maybe folks don't read help text. As is, I don't suppose the setting is too hard to discover so I'll not argue that point any further.

**Review by stokesman — APPROVED** [review]
Thanks for this Joen!

**Comment by jasmussen** [comment]
Thanks as always, @stokesman, much appreciate your time.

> It surprises me that the feature could be easily misunderstood. Especially given the help text. Maybe folks don't read help text. As is, I don't suppose the setting is too hard to discover so I'll not argue that point any further.

I'm projecting. Let me rephrase: every time I see the option I have to remind myself what exactly it does and how it works. That personal failure to consistently grok it is the primary driver for why I think there remains an opportunity to polish that one. That's all.
